app/models/name/synonymy.rb

```diff
@@ -142,21 +129,27 @@ def clear_synonym
   #   after 2:  1, 2, 3
   #
   def merge_synonyms(name)
-    # Other name has no synonyms, just transfer it over.
-    if !name.synonym_id
-      transfer_synonym(name)
-
-    # *This* name has no synonyms, transfer us over to it.
-    elsif !synonym_id
-      name.transfer_synonym(self)
-
-    # Both have synonyms -- merge them.
-    # (Make sure they aren't already synonymized!)
-    elsif synonym_id != name.synonym_id
-      name.synonyms.each { |n| transfer_synonym(n) }
+    if !self.synonym && !name.synonym
```

Comment:
Would it be at all possible to disable this cop?  I think having the unnecessary "self" in this method makes it much easier to understand.

Reply:
Sorry, no, I mean just for this method.  Also, I'm really struggling to find a way to simplify this method any more.  I feel like splitting it up will just make it harder to understand.  So I would like to request to disable the ABC cop for this method, too.  I think I was able to fix the other ABC violation (in NameControlle#check_for_new_synonym), so this may be the only one.  Note that it was already in violation of that cop, so it's not strictly "new".  And this version is, I think, much easier to understand than the old one...